keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClassicLoadBalancerHandler.kt

```diff
@@ -0,0 +1,218 @@
+    "ec2:loadBalancer:${spec.loadBalancerType}:${spec.location.accountName}:${spec.location.region}:" +
+      spec.moniker.name
+  )
+
+  override fun current(resource: Resource<ClassicLoadBalancer>): ClassicLoadBalancer? =
+    cloudDriverService.getClassicLoadBalancer(resource.spec)
+
+  override fun upsert(
+    resource: Resource<ClassicLoadBalancer>,
+    resourceDiff: ResourceDiff<ClassicLoadBalancer>
+  ): List<TaskRef> {
+    val action = when {
+      resourceDiff.current == null -> "Creating"
+      else -> "Upserting"
+    }
+
+    val taskRef = runBlocking {
+      resource.spec.let { spec ->
+        orcaService
+          .orchestrate(
+            OrchestrationRequest(
+              "$action ${spec.loadBalancerType.toString().toLowerCase()} load balancer ${spec.moniker.name} in " +
+                "${spec.location.accountName}/${spec.location.region}",
+              spec.moniker.app,
+              "$action ${spec.loadBalancerType.toString().toLowerCase()} load balancer ${spec.moniker.name} in " +
+                "${spec.location.accountName}/${spec.location.region}",
+              listOf(spec.toCreateJob()),
+              OrchestrationTrigger(resource.metadata.name.toString())
+            )
+          )
+          .await()
+      }
+    }
+
+    log.info("Started task ${taskRef.ref} to create classicLoadBalancer ${resource.spec.moniker.name} in " +
+      "${resource.spec.location.accountName}/${resource.spec.location.region}")
+    return listOf(TaskRef(taskRef.ref))
+  }
+
+  override fun delete(resource: Resource<ClassicLoadBalancer>) {
+    val taskRef = runBlocking {
+      resource.spec.let { spec ->
+        orcaService
+          .orchestrate(
+            OrchestrationRequest(
+              "Delete load balancer ${spec.moniker.name} in ${spec.location.accountName}/${spec.location.region}",
+              spec.moniker.app,
+              "Delete load balancer ${spec.moniker.name} in ${spec.location.accountName}/${spec.location.region}",
+              listOf(spec.toDeleteJob()),
+              OrchestrationTrigger(resource.metadata.name.toString())
+            )
+          )
+          .await()
+      }
+    }
+
+    log.info("Started task ${taskRef.ref} to delete classicLoadBalancer ${resource.spec.moniker.name} in " +
+      "${resource.spec.location.accountName}/${resource.spec.location.region}")
+  }
+
+  override fun actuationInProgress(name: ResourceName) =
+    runBlocking {
+      orcaService.getCorrelatedExecutions(name.value).await()
+    }.isNotEmpty()
+
+  private fun CloudDriverService.getClassicLoadBalancer(spec: ClassicLoadBalancer): ClassicLoadBalancer? =
+    runBlocking {
+      try {
+        getClassicLoadBalancer(
+          CLOUD_PROVIDER,
+          spec.location.accountName,
+          spec.location.region,
+          spec.moniker.name
+        )
+          .await()
+          .firstOrNull()
+          .run {
+            if (this == null) {
```

Comment:
I think there's some fancier kotlin you can do with the `?` and maybe `.let` so you don't have to have this first null check.